### PR TITLE
Timelock-bump

### PR DESCRIPTION
**Pull request by SwayStar123**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)

## Changes

The following changes have been made:

- Update timelock app to use forc 0.46.0

### CODE REVIEW

File: timelock/project/contracts/timelock-contract/src/utils.sw [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
 
 use ::data_structures::Asset;
-use std::{bytes::Bytes, hash::sha256};
+use std::{bytes::Bytes, hash::{Hash, Hasher, sha256}};
+
+impl Hash for Option<Bytes> {
+    fn hash(self, ref mut state: Hasher) {
+        match self {
+            Some(bytes) => bytes.hash(state),
+            None => 0.hash(state),
```
Comment (Braqzen) [inline timelock/project/contracts/timelock-contract/src/utils.sw:10]: Unsure about the `0`s here.
I assume that hashing 0 is the equivalent of None somehow and that it's also akin to multiplying by 0. If that is the case then lets save on costs by returning ZERO_B256? Or does that not work?

Comment (Braqzen) [reply timelock/project/contracts/timelock-contract/src/utils.sw:10]: Secondly, We ought to have this implementation in the std in the future

**Review by Braqzen — COMMENTED** [review]
Readme versions

File: timelock/project/contracts/timelock-contract/src/utils.sw (hunk 1 above)
Comment (SwayStar123) [reply timelock/project/contracts/timelock-contract/src/utils.sw:10]: The hash of 0 is not 0

**Review by Braqzen — APPROVED** [review]

**Pull request merged**